disperser/apiserver/server.go

```diff
@@ -257,45 +273,73 @@ func (s *DispersalServer) disperseBlob(ctx context.Context, blob *core.Blob) (*p
 	}, nil
 }
 
-func (s *DispersalServer) getAccountRate(origin string, quorumID core.QuorumID) (*PerUserRateInfo, error) {
+type RateKeys struct {
```

Comment:
do we expect to have two different types of the allowlisted account for the same request? 
i.e. for one account that has IP A and eth address B, can they specify throughput limit with A and blob limit with B?
for simplicity, why don't we consolidate it so that it has to be from the same type?

Reply:
That's a good point. I think it makes sense to always prefer the `authenticatedAddress` if it is in the allowlist, and only check the ip address if no `authenticatedAddress` was provided or it was not found in the allowlist.